src/contracts/pods/EigenPod.sol

```diff
@@ -169,133 +140,114 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-     * @dev For more details on the Beacon Chain spec, see: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#validator
+     * @dev Create a checkpoint used to prove this pod's active validator set. Checkpoints are completed 
+     * by submitting one checkpoint proof per ACTIVE validator. During the checkpoint process, the total 
+     * change in ACTIVE validator balance is tracked, and any validators with 0 balance are marked `WITHDRAWN`.
+     * @dev Once finalized, the pod owner is awarded shares corresponding to:
+     * - the total change in their ACTIVE validator balances
+     * - any ETH in the pod not already awarded shares.
+     * @dev A checkpoint cannot be created if the pod already has an outstanding checkpoint. If
+     * this is the case, the pod owner MUST complete the existing checkpoint before starting a new one.
      */
-    function verifyBalanceUpdates(
-        uint64 oracleTimestamp,
-        uint40[] calldata validatorIndices,
-        BeaconChainProofs.StateRootProof calldata stateRootProof,
-        bytes[] calldata validatorFieldsProofs,
-        bytes32[][] calldata validatorFields
-    ) external onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_BALANCE_UPDATE) {
-        require(
-            (validatorIndices.length == validatorFieldsProofs.length) && (validatorFieldsProofs.length == validatorFields.length),
-            "EigenPod.verifyBalanceUpdates: validatorIndices and proofs must be same length"
-        );
-
-        // Balance updates should not be "stale" (older than VERIFY_BALANCE_UPDATE_WINDOW_SECONDS)
-        require(
-            oracleTimestamp + VERIFY_BALANCE_UPDATE_WINDOW_SECONDS >= block.timestamp,
-            "EigenPod.verifyBalanceUpdates: specified timestamp is too far in past"
-        );
-
-        // Verify passed-in beaconStateRoot against oracle-provided block root:
-        BeaconChainProofs.verifyStateRootAgainstLatestBlockRoot({
-            latestBlockRoot: eigenPodManager.getBlockRootAtTimestamp(oracleTimestamp),
-            beaconStateRoot: stateRootProof.beaconStateRoot,
-            stateRootProof: stateRootProof.proof
-        });
-
-        int256 sharesDeltaGwei;
-        for (uint256 i = 0; i < validatorIndices.length; i++) {
-            sharesDeltaGwei += _verifyBalanceUpdate(
-                oracleTimestamp,
-                validatorIndices[i],
-                stateRootProof.beaconStateRoot,
-                validatorFieldsProofs[i], // Use validator fields proof because contains the effective balance
-                validatorFields[i]
-            );
-        }
-        eigenPodManager.recordBeaconChainETHBalanceUpdate(podOwner, sharesDeltaGwei * int256(GWEI_TO_WEI));
+    function startCheckpoint()
+        external
+        onlyEigenPodOwner() 
+        onlyWhenNotPaused(PAUSED_START_CHECKPOINT) 
+        afterRestaking(uint64(block.timestamp)) /// TODO - this is the wrong condition
```

Comment:
is this incorrect because of the difference between our existing oracle and the direct usage of EIP-4788 here?

Reply:
i'm not 100% sure it's wrong, just smells funny. I need to think about what happens if you call `activateRestaking` and then try to verify WC. 

Although with the new PR to change the way `activateRestaking` works, this may be less of a problem.